impala/tests/test_http_connect.py

```diff
@@ -24,10 +24,20 @@
 from six.moves import socketserver
 
 from impala.error import HttpError
-from impala.tests.util import ImpylaTestEnv
+from impala.tests.util import ImpylaTestEnv, is_ipv6_only_host
 
 ENV = ImpylaTestEnv()
 
+IS_IPV6_ONLY_HOST = is_ipv6_only_host(ENV.host, ENV.port)
+LOCAL_HOST = "::1" if IS_IPV6_ONLY_HOST else "127.0.0.1"
+
+# socketserver.TCPServer cannot listen both on ipv4 and ipv6. Listen to ipv6
+# if the hs2-http server has only ipv6 address.
+class IPv4or6TcpServer(socketserver.TCPServer):
+  address_family = socket.AF_INET6 if IS_IPV6_ONLY_HOST else socket.AF_INET
```

Comment:
This seems weird, but it's what the [docs](https://docs.python.org/3/library/socketserver.html) say to do.

Reply:
there is some interesting discussion about this including Guido himself - it seems that this is very old code that they don't want to change/improve:
https://discuss.python.org/t/ipv6-in-socketserver/79551/8